### ISSUE TITLE
Deep Neural Network

### ISSUE THREAD

**Issue by UH-1** [accord-net/framework#98]
Hello,
I'm using Accord.NET. It's amazing, help me a lot. I'm a Computer Science's student, with this framework, I can implement many model and algorithm for classification easily and effectively. Thank you for creating this.
I'm having a problem right now. If you don't mind, please help me with this.
I have tried to create a Deep Neural Network (Deep Belief Network). I follow this: https://github.com/primaryobjects/deep-learning and create my own network.
For details: My network have 3 hidden layers - 100 neurons each (RBM - Bernoulli), for the input layer and the hidden layers, I have used Accord.NET Deep Belief Network. The output layer, I have used Softmax Function. Unsupervised training for the hidden layers with DeepBeliefNetworkLearning, after that, I have adjusted BackpropagationLearning of AForge to fit Softmax Layer and replaced Sum Squared Error with Cross Entropy.
Problem: With input of 0-1, the output is mostly right, somtimes it goes wrong. But if I had used a real input such as 0.5, 0.6, 20, 600, etc, it would have went wrong everytimes. I can't understand why.
Thank you for reading my problem.
I hope to hear the answer from you soon.

**Comment by cesarsouza** [comment]
Well, I am deeply sorry for not answering this question on time, especially after the hope to hear from me soon... 

... but I'd say that the problem was most likely because the activation functions were getting saturated very easily when faced with values outside the 0-1 range. It is almost always a good idea to normalize your inputs to be between (0, 1) or (-1,+1) before feeding them into a neural network (c.f. ["Should I normalize/standardize/rescale the data" in the Neural Network FAQ](http://www.faqs.org/faqs/ai-faq/neural-nets/part2/)).

The Accord.Neuro project is going to be replaced soon by [Keras#](https://github.com/cesarsouza/keras-sharp), which hopefully should provide more up-to-date implementations than the Deep Belief Networks and the other simple architectures that Accord.Neuro is currently restricted to.

Deep apologies for the delay in answering.

Regards,
Cesar
